common/persistence/elasticsearch/processor.go

```diff
@@ -163,40 +164,53 @@ func (p *processorImpl) hashFn(key interface{}) uint32 {
 	return hash % p.indexerConcurrency
 }
 
-// Add request to bulk, and record ack channel message in map with provided key
-func (p *processorImpl) Add(request *client.BulkableRequest, visibilityTaskKey string, ackCh chan<- bool) {
-	if cap(ackCh) < 1 {
-		panic("ackCh must be buffered channel (length should be 1 or more)")
-	}
-	sw := p.metricsClient.StartTimer(metrics.ElasticSearchVisibility, metrics.ESBulkProcessorRequestLatency)
-	ackChWithStopwatch := newAckChanWithStopwatch(ackCh, sw)
-	_, isDup, _ := p.mapToAckChan.PutOrDo(visibilityTaskKey, ackChWithStopwatch, func(key interface{}, value interface{}) error {
-		ackChWithStopwatchExisting, ok := value.(*ackChanWithStopwatch)
+// Add request to the bulk and return ack channel which will receive ack signal when request is processed.
+func (p *processorImpl) Add(request *client.BulkableRequest, visibilityTaskKey string) <-chan bool {
+	ackCh := newAckChan()
+	retCh := ackCh.ackChInternal
+	_, isDup, _ := p.mapToAckChan.PutOrDo(visibilityTaskKey, ackCh, func(key interface{}, value interface{}) error {
+		ackChExisting, ok := value.(*ackChan)
 		if !ok {
-			p.logger.Fatal(fmt.Sprintf("mapToAckChan has item of a wrong type %T (%T expected).", value, &ackChanWithStopwatch{}), tag.Value(key))
+			p.logger.Fatal(fmt.Sprintf("mapToAckChan has item of a wrong type %T (%T expected).", value, &ackChan{}), tag.Value(key))
```

Comment:
maybe return an error?

Reply:
I think it is ok. It should never ever happend. Actually event removing `,ok` part would be a better option and just let it panic with default go message.